config/Dockerfiles/rpmbuild/rpmbuild-test.sh

```diff
@@ -1,74 +1,117 @@
 popd
 
 ABIGAIL_BRANCH=$(echo ${fed_branch} | sed 's/./&c/1')
 if [ "${fed_branch}" = "master" ]; then
     ABIGAIL_BRANCH="fc27"
 fi
 # Make repo with the newly created rpm
-rm -rf ${OUTPUTDIR}/${fed_repo}_repo
-mkdir ${OUTPUTDIR}/${fed_repo}_repo
-cp ${fed_repo}/results_${fed_repo}/${VERSION}/*/*.rpm ${OUTPUTDIR}/${fed_repo}_repo/
+rm -rf ${RPMDIR}/*
+mkdir -p ${RPMDIR}
+cp ${fed_repo}/results_${fed_repo}/${VERSION}/*/*.rpm ${RPMDIR}/
 # Run rpmlint
-rpmlint ${OUTPUTDIR}/${fed_repo}_repo/ > ${OUTPUTDIR}/logs/rpmlint_out.txt
-pushd ${OUTPUTDIR}/${fed_repo}_repo && createrepo .
+rpmlint ${RPMDIR}/ > ${LOGDIR}/rpmlint_out.txt
+pushd ${RPMDIR} && createrepo .
 popd
 # Run fedabipkgdiff against the newly created rpm
 rm -rf libabigail
 git clone git://sourceware.org/git/libabigail.git
 RPM_TO_CHECK=$(find ${fed_repo}/results_${fed_repo}/${VERSION}/*/ -name "${fed_repo}-${VERSION}*" | grep -v src)
-libabigail/tools/fedabipkgdiff --from ${ABIGAIL_BRANCH} ${RPM_TO_CHECK} &> ${OUTPUTDIR}/logs/fedabipkgdiff_out.txt
+libabigail/tools/fedabipkgdiff --from ${ABIGAIL_BRANCH} ${RPM_TO_CHECK} &> ${LOGDIR}/fedabipkgdiff_out.txt
 RPM_NAME=$(basename $RPM_TO_CHECK)
-echo "package_url=http://artifacts.ci.centos.org/fedora-atomic/${fed_branch}/repo/${fed_repo}_repo/$RPM_NAME" >> ${OUTPUTDIR}/logs/package_props.txt
+echo "package_url=http://artifacts.ci.centos.org/fedora-atomic/${fed_branch}/repo/${fed_repo}_repo/$RPM_NAME" >> ${LOGDIR}/package_props.txt
 
-exit 0
+# Perform rsync to artifacts.ci.centos.org
+RSYNC_BRANCH=${fed_branch}
+if [ "${fed_branch}" = "master" ]; then
+    RSYNC_BRANCH=rawhide
+fi
+mkdir ${RSYNC_BRANCH}
+mkdir repo
+# Kill backgrounded jobs on exit
+function clean_up {
+    # Delete the rsync lock we placed
+     rsync -vr --delete $(mktemp -d)/ fedora-atomic@artifacts.ci.centos.org::fedora-atomic/${RSYNC_BRANCH}/repo/lockdir/
+}
+trap clean_up EXIT SIGHUP SIGINT SIGTERM
+# Write uuid to a lock file and store a backup
+uuidgen > file.lock
+cp file.lock uuid.saved
+while true; do
+    # Check if lock exists on remote server
+     while [[ $(rsync --ignore-existing --dry-run -avz file.lock fedora-atomic@artifacts.ci.centos.org::fedora-atomic/${RSYNC_BRANCH}/repo/lockdir) != *"file.lock"* ]]; do
+          sleep 60
+     done
+     cp uuid.saved file.lock
+    # Push lock file with uuid to remote server
+     rsync --ignore-existing -avz file.lock fedora-atomic@artifacts.ci.centos.org::fedora-atomic/${RSYNC_BRANCH}/repo/lockdir/
+    # Pull lock file back
+     rsync -avz fedora-atomic@artifacts.ci.centos.org::fedora-atomic/${RSYNC_BRANCH}/repo/lockdir/file.lock file.lock
+    # If uuid matches, we can proceed
+     if [[ $(diff file.lock uuid.saved) == "" ]]; then
+          break
+     fi
+     sleep 60
+done
+# Rsync the empty directories over first, then the repo directory
+rsync -arv ${RSYNC_BRANCH}/ fedora-atomic@artifacts.ci.centos.org::fedora-atomic
```

Comment:
I think we need a stage pipeline to test changes and not do rsync there since that should be a known quantity IMHO